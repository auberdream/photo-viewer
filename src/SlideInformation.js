import React, { Component } from 'react';

const SlideInformation = props => (
      <div className="slide-information">
        <div className="information-container">
          <p className="photo-date">
            { props.slide.date }
          </p>
          <p className="artist-name">
            { props.slide.first_name + ' ' + props.slide.last_name }
          </p>
          <p className="photo-tags">Tags:
            { props.slide.tag.map((item, key) =>
              <span key={ key }>
                { ' ' + item }
              </span>)
            }
          </p>
        </div>
      </div>
)

export default SlideInformation;
