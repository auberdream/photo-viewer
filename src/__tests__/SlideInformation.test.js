import React from 'react';
import { shallow } from 'enzyme';
import SlideInformation from '../SlideInformation';
import renderer from 'react-test-renderer';

describe('SlideInformation', () => {

  const dummyJSON = {"id": 1,
                     "first_name": "Alan",
                     "last_name": "Berryhill",
                     "date": "2018-04-17",
                     "tag": [
                       "person",
                       "outdoors"
                     ]}

  const wrapper = shallow(<SlideInformation slide={ dummyJSON } />);


  it('renders without crashing', () => {
    shallow(<SlideInformation slide={ dummyJSON } />);
  });

  it('renders three paragraphs', () => {
    expect(wrapper.find("p").length).toEqual(3)
  });

  it('renders the date of the photo', () => {
    const dateParagraph = wrapper.find("p").at(0)
    const date = "2018-04-17"
    expect(dateParagraph.contains(date)).toEqual(true)
  });

  it("renders the artist's first and last name", () => {
    const nameParagraph = wrapper.find("p").at(1);
    const name = "Alan Berryhill";
    expect(nameParagraph.contains(name)).toEqual(true);
  });

  it('renders the tags of the photo', () => {
    const tagParagraph = wrapper.find("p").at(2);
    expect(tagParagraph.contains("Tags: person, outdoors")).toEqual(true);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<SlideInformation slide={ dummyJSON } />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
