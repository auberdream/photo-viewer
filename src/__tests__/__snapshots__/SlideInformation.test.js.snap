// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideInformation renders correctly 1`] = `
<div
  className="slide-information"
>
  <div
    className="information-container"
  >
    <p
      className="photo-date"
    >
      2018-04-17
    </p>
    <p
      className="photographer-name"
    >
      Alan Berryhill
    </p>
    <p
      className="photo-tags"
    >
      Tags: person, outdoors
    </p>
  </div>
</div>
`;
